client/my-sites/themes/theme-showcase.jsx

```diff
@@ -58,6 +60,53 @@ const optionShape = PropTypes.shape( {
 	action: PropTypes.func,
 } );
 
+const bannerProps = sample( [
+	{
+		title: translate( 'Are you a photographer? An artist?' ),
+		description: translate(
+			'Explore {{b}}PHOTO BLOG{{/b}}, an elegant theme designed to showcase your visual masterpieces.',
+			{
+				components: {
+					b: <strong />,
+				},
+			}
+		),
+		action: () => {
+			alert( 'Photo Blog banner button clicked!' );
+		},
+		actionLabel: translate( 'See the theme' ),
+		backgroundColor: '#3FE6AF',
+		image: '/calypso/images/themes-banner/photo-blog.png',
+		imageTransform: 'translateY(-4.4%) translateX(17%)',
+		imageAttrs: {
+			alt: translate( 'Photo Blog Theme' ),
+			width: 390,
+		},
+	},
+	{
+		title: translate( 'Do you run a small business?' ),
+		description: translate(
+			'We know you’re crunched for time. We created the quick-setup {{b}}SMALL BUSINESS{{/b}} theme just for you.',
+			{
+				components: {
+					b: <strong />,
+				},
+			}
+		),
+		action: () => {
+			alert( 'Small Business banner button clicked!' );
+		},
+		actionLabel: translate( 'See the theme' ),
+		backgroundColor: '#3d596d',
+		image: '/calypso/images/themes-banner/small-business.png',
+		imageTransform: 'translateY(-19%) translateX(17%)',
+		imageAttrs: {
+			alt: translate( 'Small Business Theme' ),
+			width: 410,
+		},
+	},
+] );
+
```

Comment:
I think I'd prefer defining two components (let's say `PhotgraphyThemesBanner` and `SmallBizThemesBanner`) instead of two sets of props, and choosing from them :slightly_smiling_face: What do you think?

Reply:
Yeah, that works too, but in the end we need to only be able to display one, which is chosen at the time of render. If you notice, we are doing:

```
bannerProps = sample( [ spec, spec ] )
```

It's using `lodash.sample()` to get a random _banner props object_ from the array, which is then used to render the actual banner. This is done outside the Component scope, to prevent the random component being affected by re-renders. If done inside, we would se both banners being re-rendered very fast, which would look like a roulette...

What we could do in this case is give a name to the banner props, something like:

```
const photoBlogBanner = {
	...
}

const smallBusinessBanner = {
	...
}

const bannerProps = sample( [
	photoBlogBanner,
	smallBusinessBanner,
] );```

What do you think?

Reply:
Hmm, my preferred semantics would be `Banner = sample( [ PhotgraphyThemesBanner, SmallBizThemesBanner ] )`. Are you saying that won't work? (If so, I don't get why not -- could you explain? :slightly_smiling_face:)

Reply:
Ah, yeah that works too!